### ISSUE TITLE
`randomgen` and `numpy` version sensitivity

### ISSUE THREAD

**Issue by RossBoylan** [RossBoylan/mccli#18]
This issue will be a bug as soon as `randomgen` 1.24 rolls out!

As currently noted in `README.md` this package is sensitive to the exact versions of the `Python` packages `randomgen` and `numpy`.  Aside from that note, the current handling is limited to a version restriction in https://github.com/RossBoylan/mccli/blob/792ba6c2474915fc427bf0a2c8cebc7ec11c2ca8/requirements.txt#L13.

This is not future-proof.  In particular the use of `Generator` from `randomgen` was deprecated in 1.18, and it was (or will be?) [removed](https://pypi.org/project/randomgen/) in 1.24.  Using `np.random.Generator`, i.e., `numpy`'s `Generator`, is the recommended alternative.  This issue exists mostly to fix that problem, since the [current code](https://github.com/RossBoylan/mccli/blob/792ba6c2474915fc427bf0a2c8cebc7ec11c2ca8/bin/python/montecarlo.py#L13) will break with version 1.24.  As of 2022-08-5 it seems 1.23 is still the version you get by default.

 Overall, `randomgen` was a testbed, and has gradually been merged into `numpy`.  This also means that particular versions of `randomgen` only work with particular versions of `numpy`. 

Solutions
=======

Strict version enforcement
-----------------------------------

We could make the requirements file extremely precise about which versions of the 2 libraries to use, and have code that matches.  While simple to do, it may be awkward for users, and is liable to need constant updates.  And it's easy for users ignore the version restrictions and install something else.

BTAFTP
-----------
The usual `Python` approach is "better to ask forgiveness than permission".  Something like:
```python
try:
    from randomgen import Generator
except:  # might be pickier about type of exception
    from np.random import Generator
```
This code will use the `Generator` from `randomgen` if it's available, and otherwise `numpy`'s (assuming an earlier `import numpy as np`).

This not entirely consistent with recommended use of `randomgen` which deprecated `Generator` while continuing to provide it for awhile.  So perhaps an explicit test of the `randomgen` version is in order.  But that feels a little kludgey.

Reversing the order by trying to import from `numpy` first really won't work since, AFAIK, `numpy` has always provided `Generator`. So  import from `randomgen` would never happen.  But the earlier versions in `numpy` don't provide the capabilities I need.

Just Use `numpy`'s `Generator`
------------------------------------------

The `requirements.txt` version restriction limits `randomgen` to versions in which `Generator` is either deprecated or absent.  So I could just unconditionally import it from `numpy`.

It's simple but, like the first solution, vulnerable to users who ignore the version requirements.

Remember the Rest
---------------------------
Although it is not  deprecated, we also import `PCG64` from `randomgen`.  That needs review.

**Comment by RossBoylan** [comment]
`randomgen` 1.23, the current default version, also lacks `Generator` and so breaks our code.

**Comment by RossBoylan** [comment]
Resolved in v3.1, https://github.com/RossBoylan/mccli/commit/acd8c161572e694238a7c15b4f9067d8321bd8de.
27cffcc85cc56ece0b7a02601987d490662d080e was the actual commit with the fix.
